gadget/validate.go

```diff
@@ -83,29 +83,37 @@ func ruleValidateVolumes(vols map[string]*Volume, model Model, extra *Validation
 		}
 	}
 
-	expectedSeed := false
+	hasModes := false
+	isClassicWithModes := false
 	if model != nil {
-		expectedSeed = wantsSystemSeed(model)
+		hasModes = hasGrade(model)
+		isClassicWithModes = classicOrUndetermined(model) && hasModes
 	} else {
 		// if system-seed role is mentioned assume the uc20
 		// consistency rules
-		expectedSeed = roles[SystemSeed] != nil
+		hasModes = roles[SystemSeed] != nil
 	}
```

Comment:
At first sight, it's not clear why the `isClassicWithModes` flag is only set in the first branch of the `if`. Couldn't it be that `model` is `nil` and that we are on a classic system? Maybe a comment could be added to explain the logic.

(and I didn't know that we can have classic systems with a gadget, or am I misunderstanding everything here? :-) )

Reply:
"Classic with modes" is UC initramfs + classic, uses kernel and gadget snaps, and it requires a model and  modeenv. I'll add a comment to clarify.